src/builds/builds.controller.ts

```diff
@@ -70,6 +70,7 @@ export class BuildsController {
   @UseGuards(ApiGuard)
   async create(@Body() createBuildDto: CreateBuildDto): Promise<BuildDto> {
     const project = await this.projectService.findOne(createBuildDto.project);
+    await this.buildsService.deleteOldBuilds(project.id, project.maxBuildAllowed);
```

Comment:
not sure if this is the best place to make it
what about scheduled task?
https://docs.nestjs.com/techniques/task-scheduling

Reply:
Assuming most people are used to how Jenkins job deletes old builds on a build trigger, I had this behavior in mind and implemented it like that. If we are to implement some other mechanism, we can always have a relook with a future pull request. But, I think I would be happy with Jenkins-like behavior. Let me know what you think.

Reply:
ok, make sense